src/test/groovy/com/vacation_bot/vacation/VacationIntegrationTest.groovy

```diff
@@ -0,0 +1,160 @@
+ * Integration level test of {@Link VacationGateway}.
+ */
+class VacationIntegrationTest extends AbstractSpockIntegrationTest {
+
+    def restTemplate = new TestRestTemplate()
+
+    def headers = new HttpHeaders(accept: Arrays.asList(MediaType.APPLICATION_JSON))
+
+    String userId = UUID.randomUUID()
+    def userName = 'Alex'
+    def aliases = ['Batman', 'Joker', 'Superman']
+    @Shared
+    def startDate = '2017-10-02'
+    @Shared
+    def endDate = '2017-10-26'
+
+    @Shared
+    def parsedStartDate = LocalDate.parse(startDate)
+    @Shared
+    def parsedEndDate = LocalDate.parse(endDate)
+    @Shared
+    String period = ChronoUnit.DAYS.between(parsedStartDate, parsedEndDate)
+
+    def validUser = new UserModel(id: userId, name: userName, aliases: aliases)
+    def inputData = new VacationRequestBody(userName: userName, startDate: startDate, endDate: endDate)
+
+    @Shared
+    def static final defaultVacationTotalDays = 20
+
+    def vacationTotalDays = 18
+    def vacationTotalYear = 2017
+    def validVacationTotal = new VacationTotalModel(userId: userId, vacationTotal: vacationTotalDays, year: vacationTotalYear)
+
+    def 'exercise controller (vacation total is null)'() {
+
+        given: 'the user is saved to the db'
+        userRepository.save(validUser)
+
+        and: 'set new end date'
+        inputData.endDate = newEndDate
+
+        when: 'the controller is called'
+        String result = getResponse(inputData).getBody()
+
+        then: 'the result is correct'
+        result == expectedResult
+        def createdVacationTotal = vacationTotalRepository.findAll().stream().filter { e -> e.userId == userId && e.year == vacationTotalYear }.findFirst().get()
+        createdVacationTotal
+        createdVacationTotal.vacationTotal == expectedVacationTotalDays
+
+        def createdVacation = vacationModelRepository.findAll().stream().filter { e -> e.userId == userId }.findFirst()
+        def createdVacationPresent = createdVacation.isPresent()
+        createdVacationPresent == expectedCreatedVacation
+//        with(createdVacationPresent) {
+//            LocalDateTime.ofInstant(Instant.ofEpochMilli(createdVacation.get().startDate), ZoneOffset.UTC).toLocalDate() == parsedStartDate
+//            LocalDateTime.ofInstant(Instant.ofEpochMilli(createdVacation.get().endDate), ZoneOffset.UTC).toLocalDate() == LocalDate.parse(newEndDate)
+//        }
+
+        where:
+        newEndDate      ||  expectedCreatedVacation     ||  expectedVacationTotalDays                                                                           ||  expectedResult
+        '2017-10-15'    ||  true                        ||  defaultVacationTotalDays - ChronoUnit.DAYS.between(parsedStartDate, LocalDate.parse(newEndDate))    ||  "The registration of your vacation from ${startDate} to ${newEndDate} was successfully completed! You have left ${defaultVacationTotalDays - ChronoUnit.DAYS.between(parsedStartDate, LocalDate.parse(newEndDate))} days"
```

Comment:
You can hardcode `expectedVacationTotalDays` for tests. Then the test will be more clean.